packages/itwin/grouping-mapping-widget/src/components/Mappings/Import/SelectITwin.tsx

```diff
@@ -76,43 +64,37 @@ const SelectITwin = ({
 }: SelectITwinProps) => {
   const { getAccessToken } = useGroupingMappingApiConfig();
   const iTwinsClient = useITwinsClient();
-  const [isLoading, setIsLoading] = useState<boolean>(true);
-  const [iTwins, setITwins] = useState<ITwin[]>([]);
   const [iTwinType, setITwinType] = useState<number>(defaultITwinType);
 
-  useEffect(() => {
-    void fetchITwins(setITwins, setIsLoading, getAccessToken, iTwinsClient, iTwinType);
-  }, [getAccessToken, iTwinsClient, setIsLoading, iTwinType]);
+  const { data: iTwins, isFetching: isLoading } = useQuery({
+    queryKey: ["iTwin", iTwinType],
```

Comment:
`iTwinsByType` instead of `iTwin`. What if you need to fetch a single iTwin? Of course it will have different value (`iTwinType`) but still - I think it would be better to keep these buckets different.

Reply:
Good point. Renamed the key.